src/pages/index.js

```diff
@@ -1,60 +1,123 @@
 import "./index.css";
-
+import Api from "../components/Api.js";
 import Card from "../components/Card.js";
 import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
-import {initialCards, selectors, validationSettings, profileEditForm, addCardForm, nameInput, aboutInput} from "../utils/constants.js";
+import { selectors, validationSettings, profileEditForm, addCardForm, editAvatarForm, nameInput, aboutInput } from "../utils/constants.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation";
+
+// Initialize api
+const api = new Api({baseUrl: "https://around-api.en.tripleten-services.com/v1", authToken: "9ef0145a-4550-4d8d-9dc2-252c6d573e29"});
+
+
+// create delete card popup
+const deleteCardPopup = new PopupWithConfirmation({
+  popupSelector: selectors.deleteCardPopup,
+  handleConfirm: (cardId) => {
+    api.deleteCard(cardId);
+  }
+});
 
 // Card renderer function
 const renderCard = (data) => {
   const cardElement = new Card(
     {
-      data,
+      data: data,
       handleImageClick: (imgData) => {
         cardPreviewPopup.open(imgData);
-      }
+      },
+      handleDelete: (card) => {
+        deleteCardPopup.open(card);
+        deleteCardPopup.setSubmitAction((card) => {
+          api.deleteCard(card.getId());
+          card.removeCard();
+        });
+      },
+      confirmPopup: deleteCardPopup,
+      api: api
     },
     selectors.cardTemplate
   );
   cardSection.addItem(cardElement.generateCard());
 }
 
 
+
+// Get initial user information and cards
+api.getUserInfo()
+  .then(res => {
+    document.querySelector(".profile__title").textContent = res.name;
+    document.querySelector(".profile__description").textContent = res.about;
+    document.querySelector(".profile__image").src = res.avatar;
+  });
+
+api.getInitialCards()
```

Comment:
You need to combine the requests of the user's info and the cards together in `Promise.all` because it might happen so that the `_id` has not been received yet but the cards have and your cards will be rendered not correctly. 
More info is here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

```js

Promise.all([getUserInfo(), getCards()])
// destructure the response
  .then(([userData, cards]) => {
      // set all the data
  })
  .catch(err => {
    // catch possible errors
  });

```